File: Algorithms/C++/Minimum_Amount_of_Time_to_Collect_Garbage/Minimum_Amount_of_Time_to_Collect_Garbage.cpp
#include <typeinfo>

class Solution {
public:
    int garbageCollection(vector<string>& garbage, vector<int>& travel) {
        int m, p, g, index, output, count;
        m = p = g = index = output = count = 0;
        for(auto it = garbage.begin(); it != garbage.end(); ++it){
            for(auto it2 = it->begin(); it2 != it->end(); ++it2){
                if(*it2 == 'M'){
                    m++;
                    index = count;
                }
            }
            count++;
        }
        if(m != 0){
                for(int i =  0; i < index; i ++){
                    output += travel[i];
                }
        }
        output += m;
        count = 0;
        for(auto it = garbage.begin(); it != garbage.end(); ++it){
            // cout << *it << endl;
            for(auto it2 = it->begin(); it2 != it->end(); ++it2){
                if(*it2 == 'P'){
                    p++;
                    index = count;
                }
            }
            count++;
        }   
        if(p != 0){
                for(int i =  0; i < index; i ++){
                    output += travel[i];
                }
        }
        output += p;
        count = 0;
        for(auto it = garbage.begin(); it != garbage.end(); ++it){
            // cout << *it << endl;
            for(auto it2 = it->begin(); it2 != it->end(); ++it2){
                if(*it2 == 'G'){
                    g++;
                    index = count;
                }
            }
            count++;
        }
        if(g != 0){
                for(int i =  0; i < index; i ++){
                    output += travel[i];
                }
        }
        output += g;
        return output;
    }
};

File: Algorithms/Python/Minimum_Amount_of_Time_to_Collect_Garbage/Minimum_Amount_of_Time_to_Collect_Garbage.py
class Solution:
    def garbageCollection(self, garbage: List[str], travel: List[int]) -> int:
        m = p = g = index = output = 0
        for i, w in enumerate(garbage):
            if "M" in w:
                for c in w:
                    if c == "M":
                        m = m + 1
                index = i
        if m != 0:
            output += m
            for i in range(index):
                output += travel[i]
        index = 0
        print(output)
        for i, w in enumerate(garbage):
            if "P" in w:
                for c in w:
                    if c == "P":
                        p = p + 1
                index = i
        if p != 0:
            output += p
            for i in range(index):
                output += travel[i]
        index = 0
        print(output)
        for i, w in enumerate(garbage):
            if "G" in w:
                for c in w:
                    if c == "G":
                        g = g + 1
                index = i
        if g != 0:
            output += g
            for i in range(index):
                output += travel[i]
        print(output)
        return output






    
        
    

